pennylane/templates/subroutines/flip_sign.py

```diff
@@ -0,0 +1,147 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the FlipSign template.
+"""
+
+import numpy as np
+import pennylane as qml
+from pennylane.operation import Operation, AnyWires
+
+
+class FlipSign(Operation):
+    r"""FlipSign operator flips the sign for a given basic state.
+
+    In a nutshell, this class perform the following operation:
+
+    FlipSign(n):math:`|m\rangle = -|m\rangle` if m = n
+    FlipSign(n):math:`|m\rangle = |m\rangle` if m != n
+
+    Where n is the basic state to flip and m is the input.
+    It flips the sign of the state.
+
+    Args:
+        wires (array[int]): wires that the operator acts on
+        n (array[int]) or int: binary array vector or integer value representing the state to flip the sign
+
+    Raises:
+        ValueError: "expected at integer greater than zero for basic flipping state "
+        ValueError: "expected at integer binary array "
+        ValueError: "expected at integer array for wires "
+        ValueError: "expected at least one wire representing the qubit "
+
+    .. seealso:: :func:`~.relevant_func`, :class:`~.RelevantClass` (optional)
+
+    .. details::
+
+        :title: Usage Details
+
+        The template is used inside a qnode.
+        The number of shots has to be explicitly set on the device when using sample-based measurements:
+
+        .. code-block:: python
+
+            dev = qml.device("default.qubit", wires=5, shots = 1000)
+
+            @qml.qnode(dev)
+            def circuit():
+               for wire in list(range(2)):
+                    qml.Hadamard(wires = wire)
+               qml.FlipSign([1,0], wires = list(range(2)))
+               return qml.sample()
+
+            circuit()
+
+        The result for the above circuit is:
+
+            >>> print(drawer())
+            0: ──H─╭FlipSign──H─┤  Sample
+            1: ──H─├FlipSign──H─┤  Sample
+            2: ──H─├FlipSign──H─┤  Sample
+            3: ──H─├FlipSign──H─┤  Sample
+            4: ──H─╰FlipSign──H─┤  Sample
+
+    """
+
+    num_wires = AnyWires
+
+    def __init__(self, n, wires, do_queue=True, id=None):
+
+        if type(n) == int:
+            if n > 0:
+                n = self.to_list(n)
+            else:
+                raise ValueError("expected at integer greater than zero for basic flipping state ")
+
+        if np.array(n).dtype != np.dtype("int"):
+            raise ValueError("expected at integer binary array ")
+
+        if not isinstance(wires, list):
+            raise ValueError("expected at integer array for wires ")
+
+        if np.array(wires).dtype != np.dtype("int"):
+            raise ValueError("expected a integer array for wires ")
+
+        if len(wires) == 0:
+            raise ValueError("expected at least one wire representing the qubit ")
+
+        self._hyperparameters = {"n": n}
+        super().__init__(wires=wires, do_queue=do_queue, id=id)
+
+    @staticmethod
+    def to_list(n):
```

Comment:
Here you have to pass on how many wires you are using.
Possible idea:

```python
def to_list(n, n_wires):
        r"""Convert an integer into a binary integer list"""
        
        # ! an error should be added here to check that it is possible to encode n with n_wires wires
        # ! This is check that: n < 2 ** n_wires
        
        bin_string = f"{n:b}".zfill(n_wires)
        bin_n = [int(i) for i in bin_string]
        return bin_n
```

Reply:
I agree but If I do that I should put the call to this function after checking the wires' exceptions

Reply:
Also, here I have a question, the exception raises when n **>** 2 ** n_wires?

Reply:
when n >= 2** n_wires 💪

Reply:
good!